### ISSUE TITLE
Password and Secrets: false positive using Secrets manager

### ISSUE THREAD

**Issue by uripb** [Checkmarx/kics#5042]
In a CDK project, when a "secret" is obtained by the "secretValueFromJson" method, it adds a reference to the secret inside the CloudFormation. When we analyze it with Kics we get:

Passwords And Secrets - Generic Secret, Severity: HIGH
Description: Query to find passwords and secrets in infrastructure code.
Platform: Common

We are using Kics v1.5.3.

CDK code exemple:
```typescript
const templatedSecret = new secretsmanager.Secret(this, 'TemplatedSecret', {
  generateSecretString: {
    secretStringTemplate: JSON.stringify({ username: 'user' }),
    generateStringKey: 'password',
  },
});


const connProps = {
    CONNECTION_URL: `mongodb://${endpoint}/${databaseName}`,
    USERNAME: templatedSecret.secretValueFromJson('username').toString(),
    PASSWORD: templatedSecret.secretValueFromJson('password').toString(),
    JDBC_ENFORCE_SSL: true
};


const conn = new CfnConnection(scope, `DocDB Connection`, {
       ...
      connectionProperties: connProps
    }
});
```

CloudFormation output:

```yaml
...
    Type: AWS::Glue::Connection
    Properties:
      CatalogId: "1111111111111"
      ConnectionInput:
        ConnectionProperties:
          CONNECTION_URL:
            Fn::Join:
              - ""
              - - "mongodb://{{resolve:secretsmanager:arn:"
                - Ref: AWS::Partition
                - :secretsmanager:eu-west-1:1111111111111:secret:/test/resources/docdb-test:SecretString:endpoint::}}/test
          USERNAME:
            Fn::Join:
              - ""
              - - "{{resolve:secretsmanager:arn:"
                - Ref: AWS::Partition
                - :secretsmanager:eu-west-1:1111111111111:secret:/test/resources/docdb-test:SecretString:username::}}
          PASSWORD:
            Fn::Join:
              - ""
              - - "{{resolve:secretsmanager:arn:"
                - Ref: AWS::Partition
                - :secretsmanager:eu-west-1:1111111111111:secret:/test/resources/docdb-test:SecretString:password::}}
          JDBC_ENFORCE_SSL: true
        ConnectionType: MONGODB
...
```

Could it be a false positive? Since in the CloudFormation there is not a secret, it's a reference to it.

According to the official documentation of the AWS CDK it's correct:

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_secretsmanager-readme.html

Thanks.

**Comment by rafaela-soares** [comment]
Hi, @uripb! I hope this message finds you well!

Thank you so much for contacting us and reporting the bug 😊 

We took a look, and it can be considered a false positive. We will take care of it as soon as possible!

**Comment by rafaela-soares** [comment]
Hello again, @uripb!

We are happy to inform you that PR #5048 solves the bug! 

We used your example as a negative test sample. Hope you do not mind 😊

**Comment by uripb** [comment]
Hello!

Thank you so much! No problem that you use my example! 😊
